src/frontend/src/lib/utils/format.utils.ts

```diff
@@ -18,15 +19,33 @@ interface FormatTokenParams {
 export const formatToken = ({
 	value,
 	unitName = ETHEREUM_DEFAULT_DECIMALS,
-	displayDecimals = DEFAULT_DISPLAY_DECIMALS,
+	displayDecimals,
 	trailingZeros = false,
 	showPlusSign = false
 }: FormatTokenParams): AmountString => {
 	const res = Utils.formatUnits(value, unitName);
+
+	const match = res.match(/^0\.0*/);
+	const leadingZeros = match ? match[0].length - 2 : 0;
+
+	if (isNullish(displayDecimals) && leadingZeros >= MAX_DEFAULT_DISPLAY_DECIMALS) {
+		return '< 0.00000001';
+	}
+
+	const maxFractionDigits =
+		leadingZeros + 2 > MAX_DEFAULT_DISPLAY_DECIMALS
+			? MAX_DEFAULT_DISPLAY_DECIMALS
+			: leadingZeros + 2;
```

Comment:
do I read it correctly?

```suggestion
	const maxFractionDigits =Math.min(
		leadingZeros + 2 , MAX_DEFAULT_DISPLAY_DECIMALS)
```